### PR TITLE
MCLOUD-6327: REDIS_BACKEND should work with Magento since 2.3.0

### PR DESCRIPTION
**Pull request by BaDos**
### Description

Now **REDIS_BACKEND** should configure Redis backend model for Magento since 2.3.0

### Fixed Issues (if relevant)

1. https://jira.corp.magento.com/browse/MCLOUD-6327

### Manual testing scenarios

There is FT, but need to recheck these:
https://jira.corp.magento.com/browse/MAGECLOUD-17
https://jira.corp.magento.com/browse/MAGECLOUD-48

### Release notes

For user-facing changes, add a meaningful release note. For examples, see [Magento Cloud ECE-tools release notes](https://devdocs.magento.com/cloud/release-notes/ece-release-notes.html).

### Associated documentation updates

Add link to Magento DevDocs PR or Issue, if needed.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] Pull request introduces user-facing changes and includes meaningful updates for any required release notes and documentation changes
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by BaDos** [comment]
This PR depends on https://github.com/magento/magento-cloud-patches/pull/52

**Review by shiftedreality — DISMISSED** [review]

**Review by oshmyheliuk — DISMISSED** [review]

**Review by shiftedreality — DISMISSED** [review]

**Review by shiftedreality — DISMISSED** [review]

File: src/Test/Functional/Acceptance/RedisPhp72Cest.php [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+                    ]
+                ],
+                'buildSuccess' => false,
+                'deploySuccess' => false,
+                'errorBuildMessage' => 'The REDIS_BACKEND variable contains an invalid value TestRedisModel.'
+                    . ' Use one of the available value options: Cm_Cache_Backend_Redis,'
+                    . ' \Magento\Framework\Cache\Backend\Redis,'
+                    . ' \Magento\Framework\Cache\Backend\RemoteSynchronizedCache.',
+                'errorDeployMessage' => '',
+            ],
```
Comment (oshmyheliuk) [inline src/Test/Functional/Acceptance/RedisPhp72Cest.php:72]: This code is duplicate of code above except for version, please refactor it

**Review by oshmyheliuk — DISMISSED** [review]

**Review by oshmyheliuk — APPROVED** [review]

**Comment by kandy** [comment]
Can you please test that if \Magento\Framework\Cache\Backend\RemoteSynchronizedCache and Redis_Slave_Enabled is configured the slave configuration go to proper section (remote_backend_options)?

File: codeception.dist.yml [hunk 2]
```diff
@@ -24,10 +24,10 @@ modules:
       composer_magento_username: "%REPO_USERNAME%"
       composer_magento_password: "%REPO_PASSWORD%"
       composer_github_token: "%GITHUB_TOKEN%"
-      printOutput: false
+      printOutput: true
```
Comment (shiftedreality) [inline codeception.dist.yml:27]: Is it intended to change?

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: codeception.dist.yml (hunk 2 above)
Comment (BaDos) [reply codeception.dist.yml:27]: I changes it for testing.
Need to change back in the next PR: https://github.com/magento/ece-tools/pull/766

**Review by shiftedreality — APPROVED** [review]

File: codeception.dist.yml (hunk 2 above)
Comment (shiftedreality) [reply codeception.dist.yml:27]: Ok

Comment (BaDos) [reply codeception.dist.yml:27]: Also, I'm going to add ability to change it via env variables.

> Can you please test that if \Magento\Framework\Cache\Backend\RemoteSynchronizedCache and Redis_Slave_Enabled is configured the slave configuration go to proper section (remote_backend_options)?

I have checked it on pro env.
Looks like it works fine.

**Comment by mveeramneni** [comment]
QA Approved

**Pull request merged**